app/backend/approaches/chatreadretrieveread.py

```diff
@@ -105,7 +105,11 @@ def run(self, history: Sequence[dict[str, str]], overrides: dict[str, Any]) -> A
         
         chat_content = chat_completion.choices[0].message.content
 
-        return {"data_points": results, "answer": chat_content, "thoughts": f"Searched for:<br>{q}<br><br>Prompt:<br>" + prompt.replace('\n', '<br>')}
+        msg_to_display = ''
+        for msg in messages:
+            msg_to_display += str(msg) + '\n\n'
```

Comment:
There's a one-liner to do that in Python, if you'd like:

```
msg_to_display = '\n\n'.join(str(message) for message in messages)
```

That uses a tuple comprehension and then joins the tuple.

I see you later replace \n with br, I assume you do that later instead of using br here as you're also trying to replace \n inside the content keys?

Reply:
Updated above to use a list comprehension, more performant than what I previously had.

Reply:
Thats right, and updated with list comprehensions